src/porepy/viz/exporter.py

```diff
@@ -520,546 +1178,889 @@ def _export_grid(
+            # Identify all cells as tetrahedra.
+            cells = np.arange(g.num_cells)
+
+            # Add offset taking into account previous grids
+            cell_id += (cells + cell_offset).tolist()
+
+            # Determine local connectivity for all cells. Simplices are invariant
+            # under permuatations. Thus, no specific ordering of nodes is required.
+
+            # After all, the procedure is fairly similar to the treatment of
+            # tetra cells. Most of the following code is analogous to
+            # _simplex_cell_to_nodes(). However, for hexaedron cells the order
+            # of the nodes is important and has to be adjusted. Based on the
+            # specific definition of TensorGrids however, the node numbering
+            # can be hardcoded, which results in better performance compared
+            # to a modular procedure.
+
+            # Determine cell-node ptr
+            cn_indptr = (
+                g.cell_nodes().indptr[:-1]
+                if cells is None
+                else g.cell_nodes().indptr[cells]
+            )
+
+            # Collect the indptr to all nodes of the cell;
+            # each n-simplex cell contains n nodes
+            expanded_cn_indptr = np.vstack([cn_indptr + i for i in range(8)]).reshape(
+                -1, order="F"
+            )
+
+            # Detect all corresponding nodes by applying the expanded mask to indices
+            expanded_cn_indices = g.cell_nodes().indices[expanded_cn_indptr]
+
+            # Convert to right format.
+            cn_indices = np.reshape(expanded_cn_indices, (-1, 8), order="C")
+
+            # Reorder nodes to comply with the node numbering convention of meshio
+            # regarding hexahedron cells.
+            cn_indices = cn_indices[:, [0, 2, 6, 4, 1, 3, 7, 5]]
+
+            # Test whether the hard-coded numbering really defines a hexahedron.
+            assert self._test_hex_meshio_format(g.nodes, cn_indices)
+
+            # Store cell-node connectivity, and add offset taking into account
+            # previous grids
+            cell_to_nodes = np.vstack((cell_to_nodes, cn_indices + nodes_offset))
+
+            # Update offset
+            nodes_offset += g.num_nodes
+            cell_offset += g.num_cells
+
+        # Initialize the meshio data structure for the connectivity and cell ids.
+        # There is only one cell type.
+        meshio_cells = np.empty(1, dtype=object)
+        meshio_cell_id = np.empty(1, dtype=object)
+        meshio_cells[0] = meshio.CellBlock("hexahedron", cell_to_nodes.astype(int))
+        meshio_cell_id[0] = np.array(cell_id)
+
+        # Return final meshio data: points, cell (connectivity), cell ids
+        return Meshio_Geom(meshio_pts, meshio_cells, meshio_cell_id)
+
+    def _test_hex_meshio_format(self, nodes, cn_indices) -> bool:
+
+        import numba
+
+        @numba.jit(nopython=True)
+        def _function_to_compile(nodes, cn_indices) -> bool:
```

Comment:
Again, if this is time consuming, informing numba on data types could help a lot.